src/main/webapp/app/shared/link-preview/components/link-preview/link-preview.component.ts

```diff
@@ -1,14 +1,70 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { LinkPreview } from 'app/shared/link-preview/services/link-preview.service';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import { MetisService } from 'app/shared/metis/metis.service';
+import { Posting } from 'app/entities/metis/posting.model';
 
 @Component({
     selector: 'jhi-link-preview',
     templateUrl: './link-preview.component.html',
     styleUrls: ['./link-preview.component.scss'],
 })
-export class LinkPreviewComponent {
+export class LinkPreviewComponent implements OnInit {
     @Input() linkPreview: LinkPreview;
     @Input() showLoadingsProgress: boolean;
     @Input() loaded: boolean;
     @Input() hasError: boolean;
+    @Input() posting?: Posting;
+    @Input() isReply?: boolean;
+
+    isAuthorOfOriginalPost: boolean;
+
+    faTimes = faTimes;
+
+    constructor(private metisService: MetisService) {}
+
+    ngOnInit() {
+        this.isAuthorOfOriginalPost = this.metisService.metisUserIsAuthorOfPosting(this.posting!);
+    }
+
+    /**
+     * Removes the link preview from the list of link previews
+     *
+     * @param {LinkPreview} linkPreview the link preview to be removed
+     */
+    removeLinkPreview(linkPreview: LinkPreview) {
+        const urlToSearchFor = linkPreview.url;
+
+        // Regular expression pattern to match URLs
+        // eslint-disable-next-line no-useless-escape
+        const urlRegex = /https?:\/\/[^\s/$.?#>][^\s>]*?(?=\s|[\]\)]|$)/g;
```

Comment:
Can you combine this with Theo other occurrence into one exported variable e.g. in the linkyfy service?

Reply:
done 👍